sway/sway-output.5.scd

```diff
@@ -157,6 +157,20 @@ must be separated by one space. For example:
 	adaptive sync can improve latency, but can cause flickering on some
 	hardware.
 
+*output* <name> render_bit_depth 8|10
```

Comment:
Config design question: do we know yet how this will interact with other related options? Like KMS "max bpc" (we don't have "force bpc" or "current bpc" yet unfortunately, but it's been discussed), potential HDR options, etc?

Is the bit depth a good thing to let users configure? Are there some other more meaningful or high level config options we could let the user configure and which would dictate low-level things such as render format and KMS "max bpc"?

Maybe it's a bit too soon to ask these questions. Also, I don't know a lot about color management and HDR. If everything is blurry still, maybe we can label this with an "experimental" disclaimer, saying it can go away or change in the future.

Reply:
Some relevant comments were in https://github.com/swaywm/wlroots/pull/3162#issuecomment-913005351. I'm not sure I like the mode string extension idea too much, I'd prefer not to end up with hard-to-decipher mode strings, and instead use clearer output commands, even if multiple commands might change the hw limitations and such.

Reply:
On X11, DefaultDepth 30 just extends sRGB (or whatever profile the X11 display claims to have, I think it's a property, either way see https://www.freedesktop.org/software/colord/intro.html) all over the 30 bits. Only Chromium is being a jerk and needs to have the profile set, or it becomes super low contrast, anyway.

Reply:
I have no clue about HDR.

Reply:
Color management should not otherwise differ between 10-bit and 8-bit, I fail to see how.

Reply:
> Maybe it's a bit too soon to ask these questions. Also, I don't know a lot about color management and HDR. If everything is blurry still, maybe we can label this with an "experimental" disclaimer, saying it can go away or change in the future.

I've added a short disclaimer.

> Config design question: do we know yet how this will interact with other related options? Like KMS "max bpc" (we don't have "force bpc" or "current bpc" yet unfortunately, but it's been discussed), potential HDR options, etc?

> Is the bit depth a good thing to let users configure? Are there some other more meaningful or high level config options we could let the user configure and which would dictate low-level things such as render format and KMS "max bpc"?

I think that `output <> render_bit_depth` is a command similar to `output <> modeline`, in that most people will never need to use it, and would instead use hypothetical commands like:
```
output <name> hdr disabled|enabled
output <name> wire_bit_depth 6|8|10|12|14|16
```
In this example, `output <name> hdr` would imply a render format with 16 or 16-float bpc; `wire_bit_depth` would correspond to "force bpc", and would set the most compact render format with the necessary precision. A backend/renderer combination which supports color management might require higher bpc formats. But there would still be cases where one might want to have a 16 bit render buffer when the other settings suggest 8-bit is sufficient (for higher precision colors when screen recording (?)), or force an 8-bit render buffer despite the renderer/backend preferring 16f for color management (to use less memory and power). These are where an `output <> render_bit_depth` override might be useful.

Reply:
> which supports color management might prefer higher bpc formats

Definitely. When there is a mismatch between the content colour space and the display, any transformation will typically lose noticeable information when using the low output bit depth of 8.

I'm curious, what reason do you have in mind with screen recording?

Reply:
> I'm curious, what reason do you have in mind with screen recording?

* As mentioned in https://github.com/swaywm/wlroots/pull/3162#issuecomment-913005351, sometimes programs produce 10 or 16 bit buffers; rendering to a higher depth, and taking a screenshot with the depth, will avoid losing color information from what the program provided.
* Screen sharing: if computer A has an 8 bit monitor, but computer B has a 10 bit monitor, then when looking at computer A's screen from computer B, it would be nice to have colors available at full precision.
* The RGB->YUV conversion when recording a video from a screen often loses a very small amount of information; capturing a 10 bpc screen, instead of an 8bpc screen, can slightly improve quality.

Reply:
I see, thanks for the explanation.

Reply:
@mstoeckl, this design looks good to me, thanks!

Reply:
By the way, I've found this, it seems to explain HDR well https://www.x.org/wiki/Events/XDC2016/Program/xdc-2016-hdr.pdf